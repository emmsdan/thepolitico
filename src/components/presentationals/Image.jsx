import React from 'react';
import PropTypes from 'prop-types';

const Image = props => {
  return <img src={props.image.src} className={props.image.class} />;
};

Image.propTypes = {
  image: PropTypes.shape({
    src: PropTypes.string,
    class: PropTypes.string,
    alt: PropTypes.string,
    style: PropTypes.string,
  }),
};

export default Image;
