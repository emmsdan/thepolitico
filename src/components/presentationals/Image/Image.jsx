import React from 'react';
import PropTypes from 'prop-types';

const Image = ({ src, alt, className, style }) => {
  return (
    <img src={src} className={className || ''} alt={alt || src} style={style} />
  );
};

Image.propTypes = {
  src: PropTypes.string.isRequired,
  className: PropTypes.string,
  alt: PropTypes.string,
  style: PropTypes.string,
};

export default Image;
