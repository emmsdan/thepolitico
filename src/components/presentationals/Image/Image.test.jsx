import React from 'react';
import { shallow } from 'enzyme';

import Image from './Image';

describe('<Image />', () => {
  const image = {
    src: 'http://image-url/dsf.png',
    alt: 'flowers',
    className: '',
    style: '',
  };

  it('should render correctly', () => {
    const component = shallow(<Image {...image} />);
    expect(component).toMatchSnapshot();
  });
});
