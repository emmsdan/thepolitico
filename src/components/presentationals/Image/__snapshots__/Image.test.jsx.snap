// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render correctly 1`] = `
<img
  alt="flowers"
  className=""
  src="http://image-url/dsf.png"
  style=""
/>
`;
